glam/api/views.py

```diff
@@ -155,11 +101,89 @@ def aggregations(request):
 
         record[sub_key]["data"] = data
         response[key] = record
+    return response
 
-    if not response:
-        raise NotFound("No documents found for the given parameters")
+
+@api_view(["POST"])
+def aggregations(request):
+    """
+    Fetches aggregation data.
+
+    Expects a JSON object in the body containing the query parameters, e.g.::
+
+        {
+            "query": {
+                "channel": "nightly",
+                "probe": "gc_ms",
+                "versions": ["70"],  # OR ["70", "69", "68"]
+                "aggregationLevel": "version"  # OR "build_id"
+            }
+        }
+
+    Returns a JSON object containing the histogram data and metadata, e.g.::
+
+        {
+            "response": [
+                {
+                    "data": [
+                        {
+                            "client_agg_type": "summed-histogram",
+                            "histogram": {
+                                "0": 0.0,
+                                "1": 1920.963,
+                                ...
+                            },
+                            "percentiles": {
+                                "0": 1.0,
+                                "10": 1.0259,
+                                ...
+                            },
+                            "total_users": 1604
+                        }
+                    ],
+                    "metadata": {
+                        "build_id": null,
+                        "channel": "nightly",
+                        "metric": "gc_ms",
+                        "metric_type": "histogram-exponential",
+                        "os": "Linux",
+                        "version": "70"
+                    }
+                }
+            ]
+        }
+
+    """
+    labels_cache = caches["probe-labels"]
+    if labels_cache.get("__labels__") is None:
+        Probe.populate_labels_cache()
+
+    REQUIRED_QUERY_PARAMETERS = ["channel", "probe", "versions", "aggregationLevel"]
+    body = request.data
+
+    if body is None or body.get("query") is None:
+        raise ValidationError("Unexpected JSON body")
+
+    q = body["query"]
+
+    if any([k not in q.keys() for k in REQUIRED_QUERY_PARAMETERS]):
+        # Figure out which query parameter is missing.
+        missing = set(REQUIRED_QUERY_PARAMETERS) - set(q.keys())
+        raise ValidationError(
+            "Missing required query parameters: {}".format(", ".join(sorted(missing)))
+        )
```

Comment:
This is now both in this API view and in the Python function. We can probably remove it here and assume the `get_aggregations` handles these checks.

Reply:
oops, thanks for the check